server.go

```diff
@@ -1573,9 +1568,7 @@ func (s *server) establishConn(sp *serverPeer) error {
 	if err != nil {
 		return err
 	}
-	if err := sp.Connect(conn); err != nil {
-		return err
-	}
+	sp.Connect(conn)
```

Comment:
@dajohi thanks for the review. I'll update with a solution that gives non-contradictory logging.